cabot/cabotapp/tests/test_schedules.py

```diff
@@ -138,3 +143,163 @@ def test_calendar_long_name(self):
         officers = get_duty_officers(self.schedule, at_time=datetime(2016, 11, 7, 10, 0, 0))
         emails = [str(user.email) for user in officers]
         self.assertEqual(emails, ['longnamelongnamelongnamelongname@affirm.com'])
+
+    @patch('cabot.cabotapp.models.requests.get', fake_calendar)
+    @patch('cabot.cabotapp.models.send_schedule_problems_email.apply_async')
+    @patch('cabot.cabotapp.models.timezone.now')
+    def test_validate_schedule_no_gaps(self, fake_now, fake_send_mail):
+        fake_now.return_value = datetime(2018, 8, 15, 0, 3, 54, 598552, tzinfo=timezone.utc)
+
+        self.schedule.ical_url = 'calendar_no_gaps.ics'
+        self.schedule.fallback_officer = User.objects.filter(email='dolores@affirm.com').first()
+        self.schedule.save()
+        # update_shifts(self.schedule) and self.schedule.update_problems() automatically called by post_save handler
+        self.schedule = Schedule.objects.get(pk=self.schedule.pk)  # refresh from DB
+
+        self.assertFalse(fake_send_mail.called, "Called with: {}".format(fake_send_mail.call_args))
+        self.assertIsNone(self.schedule.problems)
+
+    @patch('cabot.cabotapp.models.requests.get', fake_calendar)
+    @patch('cabot.cabotapp.models.send_schedule_problems_email.apply_async')
+    @patch('cabot.cabotapp.models.timezone.now')
+    def test_validate_schedule_with_gaps(self, fake_now, fake_send_mail):
+        fake_now.return_value = datetime(2018, 8, 15, 0, 3, 54, 598552, tzinfo=timezone.utc)
+
+        self.schedule.ical_url = 'calendar_with_gaps.ics'
+        self.schedule.fallback_officer = User.objects.filter(email='dolores@affirm.com').first()
+        self.schedule.save()
+        # update_shifts(self.schedule) and self.schedule.update_problems() automatically called by post_save handler
+        self.schedule = Schedule.objects.get(pk=self.schedule.pk)  # refresh from DB
+
+        message = """\
+The schedule <a href="{}://{}/schedule/update/{}/">Principal</a> has some issues:
+
+There are gaps in the schedule:
+* 2018-08-17 19:00:00+00:00 to 2018-08-18 19:00:00+00:00
+* 2018-08-19 19:00:00+00:00 to 2018-08-20 19:00:00+00:00
+
+If you don't want to deal with this right now, you can temporarily silence these alerts from the page linked above."""\
+            .format(settings.WWW_SCHEME, settings.WWW_HTTP_HOST, self.schedule.pk)
+
+        self.assertTrue(fake_send_mail.called)
+        fake_send_mail.assert_called_with(('Principal', message, [u'dolores@affirm.com']))
```

Comment:
Yes! (It's only like this because I was calling an assert method on the mock that didn't actually exist, but since it was a MagicMock, it just carried on and did nothing...)